drivers/entropy/entropy_cc310.c

```diff
@@ -102,9 +104,7 @@ static const struct entropy_driver_api entropy_cc3xx_rng_api = {
 #error "No cryptocell or cryptocell_sw node labels in the devicetree"
 #endif
 
-DEVICE_AND_API_INIT(entropy_cc3xx_rng, DEVICE_NAME,
-		    &entropy_cc3xx_rng_init,
-		    NULL,
-		    NULL,
-		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
-		    &entropy_cc3xx_rng_api);
+DEVICE_DT_INST_DEFINE(0, entropy_cc3xx_rng_init, device_pm_control_nop,
```

Comment:
Something's off with this driver. It does not set `dev->name` (which is NULL), so when `sys_rand32_get()` calls `device_get_binding()` it crashes at https://github.com/nrfconnect/sdk-zephyr/blob/master/kernel/device.c#L94 (observed with `mqtt_simple` sample built for `nrf9160dk_nrf9160ns`).

Reply:
I see my failure, will fix.

Reply:
@rlubos fixed, please retest

Reply:
Is it connected with?: 
```
INFO    -  17/280 nrf52840dk_nrf52840       tests/net/socket/udp/net.socket.udp.preempt        FAILED Timeout (device 70.246s)
23:29:56  INFO    - /jenkins_workspace/workspace/t-zephyr-sanitycheck_nrf_sc_test/outcomes/results/sanitycheck_build/nrf52840dk_nrf52840/tests/net/socket/udp/net.socket.udp.preempt/handler.log
23:29:56  ERROR   - ASSERTION FAIL [(dev != ((void *)0))] @ WEST_TOPDIR/zephyr/subsys/random/rand32_entropy_device.c:26
23:29:56  Device driver for CRYPTOCELL (DT_CHOSEN_ZEPHYR_ENTROPY_LABEL) not found. Check your build configuration!
23:29:56  E: r0/a1:  0x00000004  r1/a2:  0x0000001a  r2/a3:  0x00000000
23:29:56  E: r3/a4:  0x00000000 r12/ip:  0x00000000 r14/lr:  0x0000c4dd
23:29:56  E:  xpsr:  0x61000000
23:29:56  E: Faulting instruction address (r15/pc): 0x00017680
23:29:56  E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
23:29:56  E: Current thread: 0x20000dc8 (main)
23:29:56  E: Halting system
```
I see a lot of zephyr's tests failing due to this error.

Reply:
@PerMac both tests/net/socket/udp passes for me, please recheck with the latest push.

Reply:
I think this one could use DEVICE_DT_DEFINE as well instead of jumping through hoops for the INST style. It's a single-instance driver.